source/common/Card.py

```diff
@@ -1,36 +1,38 @@
-class Card():
+class Card:
     """ This defines a playing card for use in the game
 
     Every card is an integer number and a suit (jokers have suit None and number 0)
     A card can tell you its color for rule checking
     """
-    def __init__(self, number, suit):
+    def __init__(self, number, suit, deck):
         if suit not in [None, 'Spades', 'Hearts', 'Diamonds', 'Clubs']:
             raise ValueError("Invalid Suit")
         self.suit = suit
 
         if suit is None:
-            self.number = 0 #Jokers are always 0
-        elif number not in range(1,14): #range includes start but not stop number
+            self.number = 0  # Jokers are always 0
+        elif number not in range(1, 14):  # range includes start but not stop number
             raise ValueError("Invalid card number")
         else:
             self.number = number
 
+        self.deck = int(deck)
```

Comment:
Deck shouldn't need to be an int, please confirm it is absolutely required and document why in a comment if so.

Reply:
Don't think necessary.  Removed this.